### PR TITLE
Add add-on resources and types

### PR DESCRIPTION
**Pull request by vkareh**
This adds add-on types and resources, as well as cluster add-on links. Pending merge of https://gitlab.cee.redhat.com/service/uhc-clusters-service/merge_requests/1302

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -0,0 +1,3 @@
+module gitlab.cee.redhat.com/service/ocm-api-model
+
+go 1.12
```
Comment (jhernand) [inline go.mod]: This shouldn't be part of this patch.

File: model/clusters_mgmt/v1/addons_resource.model [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+		//
+		// Default value is `100`.
+		in out Size Integer
+
+		// Search criteria.
+		//
+		// The syntax of this parameter is similar to the syntax of the _where_ clause of an
+		// SQL statement, but using the names of the attributes of the flavour instead of
+		// the names of the columns of a table. For example, in order to retrieve all the
+		// flavours with a name starting with `my`the value should be:
```
Comment (jhernand) [inline model/clusters_mgmt/v1/addons_resource.model]: s/flavour/add-on/

File: model/clusters_mgmt/v1/addons_resource.model [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+		//
+		// [source,sql]
+		// ----
+		// name like 'my%'
+		// ----
+		//
+		// If the parameter isn't provided, or if the value is empty, then all the flavours
+		// that the user has permission to see will be returned.
```
Comment (jhernand) [inline model/clusters_mgmt/v1/addons_resource.model]: s/flavour/add-on/

File: model/clusters_mgmt/v1/addons_resource.model [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+		in Search String
+
+		// Order criteria.
+		//
+		// The syntax of this parameter is similar to the syntax of the _order by_ clause of
+		// a SQL statement, but using the names of the attributes of the flavour instead of
+		// the names of the columns of a table. For example, in order to sort the flavours
+		// descending by name the value should be:
```
Comment (jhernand) [inline model/clusters_mgmt/v1/addons_resource.model]: s/flavour/add-on/

File: go.mod (hunk 1 above)
Comment (vkareh) [reply go.mod]: Removed.

File: model/clusters_mgmt/v1/addons_resource.model (hunk 2 above)
Comment (vkareh) [reply model/clusters_mgmt/v1/addons_resource.model]: copy paste error :stuck_out_tongue_closed_eyes: 
Fixed.

**Comment by vkareh** [comment]
@jhernand - I've fixed your comments and added add-ons to the root_resource.
